examples/custom_diffusion/train.py

```diff
@@ -0,0 +1,1300 @@
+                break
+
+        if accelerator.is_main_process:
+            if args.validation_prompt is not None and global_step % args.validation_steps == 0:
+                logger.info(
+                    f"Running validation... \n Generating {args.num_validation_images} images with prompt:"
+                    f" {args.validation_prompt}."
+                )
+                # create pipeline
+                pipeline = CustomDiffusionPipeline.from_pretrained(
+                    args.pretrained_model_name_or_path,
+                    unet=accelerator.unwrap_model(unet),
+                    text_encoder=accelerator.unwrap_model(text_encoder),
+                    tokenizer=tokenizer,
+                    revision=args.revision,
+                    modifier_token=args.modifier_token,
+                    modifier_token_id=modifier_token_id,
+                )
+                pipeline.scheduler = DPMSolverMultistepScheduler.from_config(pipeline.scheduler.config)
+                pipeline = pipeline.to(accelerator.device)
+                pipeline.set_progress_bar_config(disable=True)
+
+                # run inference
+                generator = torch.Generator(device=accelerator.device).manual_seed(args.seed)
+                images = [
+                    pipeline(args.validation_prompt, num_inference_steps=25, generator=generator, eta=1.).images[0]
+                    for _ in range(args.num_validation_images)
+                ]
+
+                for tracker in accelerator.trackers:
+                    if tracker.name == "tensorboard":
+                        np_images = np.stack([np.asarray(img) for img in images])
+                        tracker.writer.add_images("validation", np_images, epoch, dataformats="NHWC")
+                    if tracker.name == "wandb":
+                        tracker.log(
+                            {
+                                "validation": [
+                                    wandb.Image(image, caption=f"{i}: {args.validation_prompt}")
+                                    for i, image in enumerate(images)
+                                ]
+                            }
+                        )
+
+                del pipeline
+                torch.cuda.empty_cache()
+
+    # Save the updated weights 
+    accelerator.wait_for_everyone()
+    if accelerator.is_main_process:
+        unet = unet.to(torch.float32)
+        pipeline = CustomDiffusionPipeline.from_pretrained(
+            args.pretrained_model_name_or_path,
+            unet=accelerator.unwrap_model(unet),
+            text_encoder=accelerator.unwrap_model(text_encoder),
+            tokenizer=tokenizer,
+            revision=args.revision,
+            modifier_token=args.modifier_token,
+            modifier_token_id=modifier_token_id,
+        )
+        save_path = os.path.join(args.output_dir, "delta.bin")
+        pipeline.save_pretrained(save_path, freeze_model=args.freeze_model)
```

Comment:
Should we maybe just save the trained parameters instead of the whole pipeline? This would make the output of this training much more portable. E.g. like we do here: https://github.com/huggingface/diffusers/blob/a4b233e5b5092e8ff861b5f5d3ac646fcba9ba79/examples/dreambooth/train_dreambooth_lora.py#L973

Reply:
Yes, currently the CustomDiffusionPipeline saves only trained parameters. But based on later comments, I will re-organize the training script to reuse the existing LoRA and Textual Inversion pipelines. Thanks.